src/Point.ts

```diff
@@ -0,0 +1,46 @@
+const defaultPoint = {
+    x: 0,
+    y: 0
+}
+
+export class Point {
+    public x: number;
+    public y: number;
+
+    constructor();
+    constructor(x: number, y: number);
+    constructor(x?: number, y?: number) {
```

Comment:
You can use parameters value by default like this:

```
constructor(x: number = 0,y: number = 0) {
        this.x = x;
        this.y = y;
}
```
Or even so I believe, because Typescript can assign properties, which have access modificators, on its own.

```
constructor(public x: number = 0, public y: number = 0) {
}
```